tcav/utils.py

```diff
@@ -222,3 +231,35 @@ def is_random_concept(concept):
 def make_dir_if_not_exists(directory):
   if not tf.gfile.Exists(directory):
     tf.gfile.MakeDirs(directory)
+
+
+def result_to_proto(result):
+  """Given a TCAV results dict, convert to TCAVResults.
+
+  Args:
+    result: a dictionary returned by TCAV._run_single_set()
+
+  Returns:
+    TCAVResult proto
+  """
+  result_proto = Result()
+  for key in _KEYS:
+    setattr(result_proto, key, result[key])
+  positive_set_name = result["cav_concept"]
+  negative_set_name = result["negative_concept"]
+  for val in result["val_directional_dirs"]:
+    result_proto.val_directional_dirs.append(val)
+  result_proto.cav_accuracies.positive_set_accuracy = result["cav_accuracies"][
+      positive_set_name]
+  result_proto.cav_accuracies.negative_set_accuracy = result["cav_accuracies"][
+      negative_set_name]
+  result_proto.cav_accuracies.overall_accuracy = result["cav_accuracies"][
+      "overall"]
+  return result_proto
+
+
```

Comment:
add unit test to utils_test to test this new method results_to_proto

Reply:
done. I'm unsure how to run googletest - I tried running `python utils_test.py` directly from the command line but it failed, and the googletest repo is for C++. Is there documentation for how I can run Python googletest on my local machine?

Reply:
Best way to test is to build the tcav pip package on your local machine with these changes, pip install it, then run the test. 

Build pip package: python setup.py bdist_wheel --python-tag py3 (or py2 if using python v2)
Install pip package: pip install [location of created whl file from last step] - make sure to uninstall existing tcav package first.
Run UT: python -m tcav.utils_test

Reply:
I ran the aforementioned steps, but received a `from tcav import utils; ImportError: cannot import name utils` message. I'm unsure why this is happening since `utils.py` is in the same directory as `utils_test.py`, and the entire `tcav` package should be installed under pip.

Sorry if I'm overlooking something obvious, I'm not very familiar with how pip handles dependencies like this.

Reply:
which line in which file had the error?

Reply:
it was line 27 of cav.py

Reply:
On further inspection, this appears to be a circular dependency. `utils_test` calls `utils`, which calls `results_pb2`, which calls `tcav`, which calls `cav`, which calls `utils`. I'm unsure of how `results_pb2` is calling `tcav`, since the proto should be self-contained.

Reply:
Did you last commit fix the circular dep issue? If so, has the UT passed? Thanks.

Reply:
No, the circular dependency is still there. It's caused because results_pb2.py calls tcav.py, which is behavior I didn't expect to happen

Reply:
i think its due to the package setting in the results.proto (package: tcav.tcav). Try changing it to tcav.results or similar

Reply:
I played around with the results.proto package name, and it doesn't seem to have an effect on the error. I believe the proto package name is separate from the python package name.